pkg/client/unversioned/autoscaling.go

```diff
@@ -66,12 +66,10 @@ func setAutoscalingDefaults(config *restclient.Config) error {
 	if config.UserAgent == "" {
 		config.UserAgent = restclient.DefaultKubernetesUserAgent()
 	}
-	// TODO: Unconditionally set the config.Version, until we fix the config.
-	//if config.Version == "" {
-	copyGroupVersion := g.GroupVersion
-	config.GroupVersion = &copyGroupVersion
-	//}
-
+	if config.GroupVersion == nil || !isEnabledVersion(autoscaling.GroupName, config.GroupVersion) {
```

Comment:
For all of these blocks you need to move the `g, err := ...` logic inside of this block.